src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -83,7 +83,7 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
             if (LambdaUtilities.IsLambdaBody(syntax))
             {
                 Debug.Assert(allowLambda);
-                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax);
+                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax || syntax is ArrowExpressionClauseSyntax);
```

Comment:
As you can actually see below in the ```//expression body``` check.

---
In reply to: [132549667](https://github.com/dotnet/roslyn/pull/21396#discussion_r132549667) [](ancestors = 132549667,132526663)